.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,146 @@
+# Pull Request Template
+
+## Description
+
+Please include a summary of the changes and the related issue. Provide relevant motivation and context.
+
+## How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Include any relevant details for your test configuration.
+
+- **Test Cases:**
+  - [ ] Unit tests
+  - [ ] Integration tests
+  - [ ] Manual tests
+  - [ ] No tests required
+
+## Type of Change
+
+- [ ] Bug fix
+- [ ] New feature
+- [ ] Breaking change
+- [ ] Documentation update
+
+## Checklist
+
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
```

Comment:
nit: "New and existing tests pass locally with my changes"

All new and existing tests should pass, not just unit tests